### ISSUE TITLE
The Creators element in xsd is different of documentation

### ISSUE THREAD

**Issue by tomfris** [openaire/guidelines-cris-managers#142]
We are trying to deliver datasets to openaire from FRIS , the flemish research information portal (researchportal.be). We have problems delivering the creator -element. 

The Creators element in xsd is different of documentation, as you can see in the snippet below the Creators contains a sequence of “cfLinkWithDisplayNameToPersonWithAffiliationsOrOrgUnit__Type”
1<xs:element cflink:container="ordered" minOccurs="0" name="Creators">
2  <xs:annotation>
3    <xs:documentation>The creators of this product</xs:documentation>
4  </xs:annotation>
5  <xs:complexType>
6  <xs:sequence>
7    <xs:element cflink:link=[https://w3id.org/cerif/model#Person_ResultProduct(https://w3id.org/cerif/vocab/PersonOutputContributions#Creator) https://w3id.org/cerif/model#OrganisationUnit_ResultProduct(https://w3id.org/cerif/vocab/OrganisationOutputContributions#Creator)](https://w3id.org/cerif/model#Person_ResultProduct(https://w3id.org/cerif/vocab/PersonOutputContributions#Creator)%20https://w3id.org/cerif/model#OrganisationUnit_ResultProduct(https://w3id.org/cerif/vocab/OrganisationOutputContributions#Creator)) maxOccurs="unbounded" minOccurs="0" name="Creator" type="cfLinkWithDisplayNameToPersonWithAffiliationsOrOrgUnit__Type"/>
8  </xs:sequence>
9  </xs:complexType>
10</xs:element>
 
Below you can see the cfLinkWithDisplayNameToPersonWithAffiliationsOrOrgUnit__Type structure, it’s based on cfLinkWithDisplayName__BaseType
 
1<xs:complexType name="cfLinkWithDisplayNameToPersonWithAffiliationsOrOrgUnit__Type">
2           <xs:complexContent>
3                 <xs:extension base="cfLinkWithDisplayName__BaseType">
4                       <xs:choice>
5                             <xs:group ref="PersonWithAffiliations__Group"/>
6                             <xs:element ref="OrgUnit__SubstitutionGroupHead"/>
7                       </xs:choice>
8                 </xs:extension>
9           </xs:complexContent>
10    </xs:complexType>

And the cfLinkWithDisplayName__BaseType contains only “DisplayName”
1<xs:complexType name="cfLinkWithDisplayName__BaseType">
2           <xs:complexContent>
3                 <xs:extension base="cfLink__BaseType">
4                       <xs:sequence>
5                             <xs:element minOccurs="0" name="DisplayName" type="xs:string"/>
6                       </xs:sequence>
7                 </xs:extension>
8           </xs:complexContent>
9     </xs:complexType>

So it’s not possible to create a person object with "FamilyNames” and “FirstNames” like is described in documentation:
1<Person id="Persons/21234512">
2  <PersonName>
3    <FamilyNames>Singhal</FamilyNames>
4    <FirstNames>Sonal</FirstNames>
5  </PersonName>
6</Person>

You can check further using xsd provided by openaire
[guidelines-cris-managers/openaire-cerif-profile.xsd at master · openaire/guidelines-cris-managers](https://github.com/openaire/guidelines-cris-managers/blob/master/schemas/openaire-cerif-profile.xsd) 

Can you please tell us how to fix this?

**Comment by jdvorak001** [comment]
@tomfris, would you please have an error message from the validator (or are you using some other XML parser)?

**Comment by tomfris** [comment]
No, we don't have any errors in the validator, because we are not building the creator object, since we don't know how.

We are using the openaire validator: https://github.com/EuroCRIS/openaire-cris-validator, but there everything passes. 

You can have a look at our endpoint : https://app.r4.researchportal.be/interface/oai?verb=ListRecords&metadataPrefix=oai_dc&set=openaire_cris_products.

**Comment by jdvorak001** [comment]
I trust it should be possible to add a `<Creators>` section following [the example](https://github.com/openaire/guidelines-cris-managers/blob/master/samples/openaire_cerif_xml_example_products.xml#L60-L77). The `<Person>` element inside each `<Creator>` is permitted by the `PersonWithAffiliations__Group`. It is the effect of the `<xs:extension>` in `cfLinkWithDisplayNameToPersonWithAffiliationsOrOrgUnit__Type` that this is supported by the Schema. Perhaps Example 2 on [this W3Schools page](https://www.w3schools.com/xml/el_extension.asp) sheds some light into it.

In case I am not guessing correctly where the problem is please email me and we could have a short call.

**Comment by tomfris** [comment]
The problem is that the person element is not present under the Creator element  : 

<Creators>
<Creator>
<Person__SubstitutionGroupHead>
<PersonName>
<FamilyNames>Doe</FamilyNames>
<FirstNames>John</FirstNames>
</PersonName>
</Person__SubstitutionGroupHead>
</Creator>
</Creators>

Are we using the wrong xsd? 

i'll mail you with a meeting invite.

BR,
Tom

**Comment by NazimeAtimene** [comment]
Hi, this still blocks us, did you manage to reproduce it with the POC that we've provided to you? 
Please let us know if you need more information.
Cheers,
Nazime

**Comment by ACz-UniBi** [comment]
Dear @tomfris , @NazimeAtimene ,

if I use the command line call, eg. `xjc  -d ./temp -verbose -xmlschema schemas/openaire-cerif-profile.xsd` to create classes,
then the _Person.java_ class is created, which seems to be missing at https://github.com/NazimeAtimene/openaire/tree/master/target/generated-sources/xmlbeans/eu/openaire/cerifProfile/x11 .

Could you elaborate more about creating the classes from _openaire_cerif_profile.xsd_ ?

Best, Andreas

**Comment by NazimeAtimene** [comment]
Hi @ACz-UniBi,
We're using JAXB module to generate the classes from the XSD.
We will try using your command to see if it's a generation issue and let you know asap.
Thanks for your help.
Regards,

**Comment by jdvorak001** [comment]
I'm not seeing any mention of a JAXB plugin to generate the sources in the pom.xml file, so am I correct in guessing you're currently generating the sources through your IDE? 
I believe the POM file would be a more fitting place if you want to get repeatable builds. A plugin like [this one](https://www.mojohaus.org/jaxb2-maven-plugin/Documentation/v3.1.0/index.html) could help.

**Comment by NazimeAtimene** [comment]
Hi @jdvorak001,
Thanks for your reply, you're right in the provided example we're using xmlbeans-maven-plugin to generate classes from provided xsd as we had issues with jaxb and JAVA 17 (solved). We're working on it now, and will provide you with more insights asap.
Thanks for your concern.
Regards,

**Comment by ACz-UniBi** [comment]
Thank you @NazimeAtimene for the note. 
Could share your solution here or in [your repo](https://github.com/NazimeAtimene/openaire) before we close the ticket? 

Thanks a lot.

**Comment by NazimeAtimene** [comment]
Hi @ACz-UniBi,
We've pushed the changes in the repo.
Kind regards,
